src/silx/io/url.py

```diff
@@ -211,6 +215,9 @@ def __check_validity(self):
                 self.__data_path is None and self.__data_slice is None
             ) or self.__data_path is not None
             self.__is_valid = slice_implies_data
+        elif self.__scheme.startswith("http"):
```

Comment:
It would be best to explicitly list valid schemes here:

```suggestion
        elif self.__scheme in ("http", "https"):
```